src/hotspot/share/gc/z/zFuture.inline.hpp

```diff
@@ -46,8 +46,8 @@ template <typename T>
 inline T ZFuture<T>::get() {
   // Wait for notification
   Thread* const thread = Thread::current();
-  if (thread->is_Java_thread()) {
-    _sema.wait_with_safepoint_check(thread->as_Java_thread());
+  if (JavaThread::cast(thread)) {
```

Comment:
This is wrong - we still need the is_Java_thread() query; and cast is not a boolean operator.

Reply:
Fixed. It's a wrong use of the method `cast`. Thanks for finding it. I re-read my patch to avoid the similar mistake.